src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1900,85 +1900,102 @@ impl Cage {
     //------------------------------------FCNTL SYSCALL------------------------------------
 
     pub fn fcntl_syscall(&self, fd: i32, cmd: i32, arg: i32) -> i32 {
-        let checkedfd = self.get_filedescriptor(fd).unwrap();
-        let mut unlocked_fd = checkedfd.write();
-        if let Some(filedesc_enum) = &mut *unlocked_fd {
-            let flags = match filedesc_enum {
-                Epoll(obj) => &mut obj.flags,
-                Pipe(obj) => &mut obj.flags,
-                Stream(obj) => &mut obj.flags,
-                File(obj) => &mut obj.flags,
-                Socket(ref mut sockfdobj) => {
-                    if cmd == F_SETFL && arg >= 0 {
-                        let sock_tmp = sockfdobj.handle.clone();
-                        let mut sockhandle = sock_tmp.write();
-
-                        if let Some(ins) = &mut sockhandle.innersocket {
-                            let fcntlret;
-                            if arg & O_NONBLOCK == O_NONBLOCK {
-                                //set for non-blocking I/O
-                                fcntlret = ins.set_nonblocking();
-                            } else {
-                                //clear non-blocking I/O
-                                fcntlret = ins.set_blocking();
-                            }
-                            if fcntlret < 0 {
-                                match Errno::from_discriminant(interface::get_errno()) {
-                                    Ok(i) => {
-                                        return syscall_error(
-                                            i,
-                                            "fcntl",
-                                            "The libc call to fcntl failed!",
-                                        );
-                                    }
-                                    Err(()) => panic!("Unknown errno value from fcntl returned!"),
-                                };
+        //if the provided file descriptor is out of bounds, get_filedescriptor returns Err(),
+        //and using unwrap on Err() causes a thread to panick
+        //instead, I propose using the 'if let' construct to be able to report an error to 
+        //the user instead of simply panicking
+        //otherwise, file descriptor table entry is stored in 'checkedfd'
+        if let Ok(checkedfd) = self.get_filedescriptor(fd) {        
```

Comment:
So I like the philosophy here, but I think you want to reverse the if statement behavior so that you error out quickly, rather than continually nesting your if statement.  This way one can much more easily read the code.

Reply:
If I write something like "if let Err() = self.get_filedescriptor(fd)", then in the "else", I will have to do pattern matching once again to extract checkedfd, which seems redundant. Should I use match instead?

Reply:
That would be an option.  

I'm not an expert, but code like [this](https://github.com/JustinCappos/fdtables/blob/main/src/dashmaparrayglobal.rs#L812C14-L882C1) looks more "correct" to me.  I'm very open to hearing from others, but certainly in Python, C, etc. it is preferred to do it in the style I've written.